server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -387,6 +387,11 @@ public BulkScorer bulkScorer(LeafReaderContext context) throws IOException {
                         return null;
                     }
                 }
+
+                @Override
+                public int count(LeafReaderContext context) throws IOException {
```

Comment:
🙂 did some test catch this?

Reply:
Actually yes!
The current CI failures are due to this override method only: https://build.ci.opensearch.org/job/gradle-check/33309/

What I don't understand yet is how only the test cases when `size: 0` is affected and not the non-zero ones.

Reply:
Failing tests in the above CI were from QueryPhaseTests.java which I have addressed and fixed now. 

Similar PR from @austintlee (Thanks Austin!) which helped catch this case was: https://github.com/opensearch-project/OpenSearch/pull/12179